.github/workflows/codeql-workflow.yml

```diff
@@ -0,0 +1,42 @@
+name: "CodeQL"
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+      - develop
+  pull_request:
+    branches:
+      - master
+      - develop
+    paths-ignore:
+      - '.github/workflows/*.yml'
+
+jobs:
+  job_build_debug_nano_s:
+    name: CodeQL Analyse of application
+    runs-on: ubuntu-latest
+
+    container:
+      image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder:latest
+
+    steps:
+      - name: Clone
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: ${{ matrix.language }}
+          queries: security-and-quality
+
+      - name: Build
+        run: |
+          make clean
+          make DEBUG=1 ALLOW_DATA=1
```

Comment:
`ALLOW_DATA=1` is not necessary here. It is used to automatically enable an option in the app for testing purposes.

Also, I think you shouldn't compile app in debug mode: code scanning is more useful for the release version.

Reply:
Done